### ISSUE TITLE
Can't seem to figure out how classAttributes work (Vue + Nuxt3)

### ISSUE THREAD

**Issue by letoast** [schoero/eslint-plugin-readable-tailwind#30]
I'm using Nuxt3 which uses Vue3 and can't seem to create a matcher for this, used as an attribute
```vue
<SomeComponent
    :classes="{ actions: 'sticky bg-gradient-to-b -bottom-[1px] pt-5 bg-dark-grey-900' }"
/>
```

Can't seem to match this either
```vue
<SomeComponent
    :classes="{ actions: cn('sticky bg-gradient-to-b -bottom-[1px] pt-5 bg-dark-grey-900')  }"
/>
```
If I move ``cn(...)`` into ``<script>``, then it matches the callee and if I turn ``:classes="`` into a static prop ``classes="`` it also works, but then of course the value is just a string instead of an object.

In fact dynamic props matching doesn't work at all...

```package.json
"eslint-plugin-readable-tailwind": "^1.4.0",
```

What would a matcher for this look like? 

Thank you for your help

**Comment by schoero** [comment]
[Vue bound classes](https://vuejs.org/guide/essentials/class-and-style) do not work with eslint-plugin-readable-tailwind yet but I have done some experiments for this in https://github.com/schoero/eslint-plugin-readable-tailwind/pull/31.

I am not really familiar with vue myself and I had some concerns that it may report too many false positives and potentially break user code. Therefore I decided not to release it yet.

The implementation in #31 is based on the [vue documentation](https://vuejs.org/guide/essentials/class-and-style)
The documentation states that classes can be `object Keys` or `strings` within arrays.

Thats why I created a `String` and `ObjectKey` matcher in #31

https://github.com/schoero/eslint-plugin-readable-tailwind/blob/2c8a617407c291dc2f69b9de664cd5a29b743b08/src/config/default-config.ts#L51-L69

Is this also a valid way to add classes in Vue? Is the `actions` property some kind of special property that allows this?

I would appreciate any help in answering these questions, as I have very limited experience with vue.

**Comment by schoero** [comment]
Thank you for the great explanation, that definitely helped me.

I have published a preview version. 

By default it matches `ObjectKeys` and `Strings` inside a `:class` or `v-bind:class` attribute. If someone uses other props for classes, the [config must be extended](https://github.com/schoero/eslint-plugin-readable-tailwind/blob/main/docs/api/defaults.md#extending-the-config).

Something like this will now also work, because of the `callees` matcher for `cn()`:

```vue
<template>
   <div :classes="{ actions: cn('sticky bg-gradient-to-b -bottom-[1px] pt-5 bg-dark-grey-900')  }"></div>
</template>
```

You can install the preview version using:

```sh
npm i eslint-plugin-readable-tailwind@vue
```

If you can provide some feedback and report any issues you encounter, I'm sure we can get this merged pretty soon.

**Comment by letoast** [comment]
Wow, thanks for the fast fix. 
First, there's an error I got from before, which is that the [documentation](https://github.com/schoero/eslint-plugin-readable-tailwind/blob/main/docs/api/defaults.md#extending-the-config) doesn't match what the code is expecting
```js
"readable-tailwind/sort-classes": ["warn", {
        attributes: [ <---- this should be "classAttributes"
         ...getDefaultClassAttributes(),
         [
            "myAttribute", [
            {
                match: MatcherType.String
            }
         ]
      ]
   ]
}]
```
The error is:
```bash
  Unexpected property "attributes". Expected properties: "callees", "classAttributes", "variables", "order", "tailwindConfig".
```

The next thing is that I'm no longer able to import 

I'm getting this, I wasn't getting this before
```js
import {
	getDefaultCallees,
	getDefaultClassAttributes,
	getDefaultVariables
} from 'eslint-plugin-readable-tailwind/api/defaults'
import { MatcherType } from 'eslint-plugin-readable-tailwind/api/types'
```

```bash
[11:59:19 PM]  ERROR  [uncaughtException] Cannot find module '/Users/ciril/projects/platform/node_modules/eslint-plugin-readable-tailwind/lib/esm/api/defaults.js' imported from /Users/ciril/projects/platform/eslint.config.js

  at finalizeResolution (node:internal/modules/esm/resolve:265:11)
  at moduleResolve (node:internal/modules/esm/resolve:933:10)
  at defaultResolve (node:internal/modules/esm/resolve:1157:11)
  at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:390:12)
  at ModuleLoader.resolve (node:internal/modules/esm/loader:359:25)
  at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:234:38)
  at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
  at link (node:internal/modules/esm/module_job:86:36)
 ```

If I don't try to import the defaults and MatcherType, then it works! :D Well at least the `cn(...)`part. I still don't know how to make it work for the first example, though :/

```vue
<SomeComponent
    :classes="{ actions: 'sticky bg-gradient-to-b -bottom-[1px] pt-5 bg-dark-grey-900' }"
/>
```

**Comment by schoero** [comment]
Those are not the kind of errors that I have expected, but thank you for testing 😄

I have released another preview version. 

```sh
npm i eslint-plugin-readable-tailwind@vue
```

It includes the following changes:

 - `api/defaults` and `api/types` are now included in the compiled library.
 - attributes are no longer case sensitive
 - it is no longer necessary to specify a matcher for both `v-bind:classes` and `:classes`. Either one matches now both variants.
 - The documentation is now corrected to `classAttributes`

**Comment by letoast** [comment]
@schoero thanks for the quick patch again 😄 

It works great! Exactly where I need it!
![image](https://github.com/schoero/eslint-plugin-readable-tailwind/assets/8151795/ff75be1e-ca28-4fb1-b34c-1007c527fe9b)

![image](https://github.com/schoero/eslint-plugin-readable-tailwind/assets/8151795/f1d5f31e-f426-475d-a3b5-8f5b6ed82b00)
